pepper-apis/src/main/java/org/broadinstitute/ddp/route/GetDsmParticipantStatusRoute.java

```diff
@@ -41,32 +42,43 @@ public GetDsmParticipantStatusRoute(DsmParticipantStatusService service) {
     public ParticipantStatusTrackingInfo handle(Request request, Response response) {
         String studyGuid = request.params(RouteConstants.PathParam.STUDY_GUID);
         String userGuid = request.params(RouteConstants.PathParam.USER_GUID);
-        if (StringUtils.isBlank(studyGuid)) {
-            LOG.warn("Study GUID is blank");
-            ResponseUtil.halt400ErrorResponse(response, ErrorCodes.MISSING_STUDY_GUID);
-        }
-        if (StringUtils.isBlank(userGuid)) {
-            LOG.warn("Participant GUID is blank");
-            ResponseUtil.halt400ErrorResponse(response, ErrorCodes.MISSING_USER_GUID);
-        }
-        DDPAuth ddpAuth = RouteUtil.getDDPAuth(request);
-        DsmCallResponse status = service.fetchParticipantStatus(studyGuid, userGuid, ddpAuth.getToken());
-        LOG.info("DSM call completed, study = {} and participant = {}, status = ", studyGuid, userGuid, status.getHttpStatus());
-        String errMsg = null;
-        if (status.getHttpStatus() == 404) {
-            errMsg = "Participant " + userGuid + " or study " + studyGuid + " not found";
+        String token = RouteUtil.getDDPAuth(request).getToken();
+
+        LOG.info("Attempting to fetch DSM participant status for {} in study {}", userGuid, studyGuid);
+
+        response.type(ContentType.APPLICATION_JSON.getMimeType());
+        return process(studyGuid, userGuid, token);
+    }
+
+    ParticipantStatusTrackingInfo process(String studyGuid, String userGuid, String token) {
```

Comment:
I think this refactor covers the same functionality. Can you double-check, @raiks2?